apps/web/src/utils/tokens.ts

```diff
@@ -4,33 +4,60 @@ import { parseBytes32String } from '@ethersproject/strings'
 import { TokenType } from '@safe-global/safe-gateway-typescript-sdk'
 import { ERC721_IDENTIFIER } from '@safe-global/utils/utils/tokens'
 import { type Erc20Token } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
+import { multicall } from '../../../../packages/utils/src/utils/multicall'
+import { type BytesLike } from 'ethers'
 
 /**
  * Fetches ERC20 token symbol and decimals from on-chain.
  * @param address address of erc20 token
  */
 export const getERC20TokenInfoOnChain = async (
-  address: string,
-): Promise<Omit<Erc20Token, 'name' | 'logoUri'> | undefined> => {
+  address: string | string[],
+): Promise<Omit<Erc20Token, 'name' | 'logoUri'>[] | undefined> => {
   const web3 = getWeb3ReadOnly()
   if (!web3) return
 
-  const erc20 = ERC20__factory.connect(address, web3)
+  let tokenAddresses = Array.isArray(address) ? address : [address]
 
-  const symbol = await erc20
-    .symbol()
-    .then((symbol) => symbol)
-    .catch((error) => parseBytes32String(error.value)) // Some older contracts use bytes32 instead of string
-    .finally(() => '')
+  const erc20_interface = ERC20__factory.createInterface()
 
-  const decimals = await erc20.decimals()
+  const calls = tokenAddresses.flatMap((address) => [
+    {
+      to: address,
+      data: erc20_interface.encodeFunctionData('symbol'),
+    },
+    {
+      to: address,
+      data: erc20_interface.encodeFunctionData('decimals'),
+    },
+  ])
 
-  return {
-    address,
-    symbol,
-    decimals: Number(decimals),
-    type: TokenType.ERC20,
+  const results = await multicall(web3, calls)
+
+  const tokenInfos: Omit<Erc20Token, 'name' | 'logoUri'>[] = []
+  for (let i = 0; i < results.length; i += 2) {
+    const address = tokenAddresses[i / 2]
+    if (!address) break
+    let symbol: string
+    try {
```

Comment:
Not sure I understand why we had to change the existing logic. Why do we have to iterate over results like this i.e. increasing by 2 and dividing by 2? Aren't there always exactly two elements in the `results` array? The first one for symbol the second one for decimals?